client/lib/happychat/connection.js

```diff
@@ -20,7 +20,10 @@ class Connection extends EventEmitter {
 				const socket = new IO( url );
 				socket
 					.once( 'connect', () => debug( 'connected' ) )
-					.on( 'init', () => resolve( socket ) )
+					.on( 'init', () => {
```

Comment:
@jordwest Since this PR is blocking a few others and it's functionally correct I'm going to merge, but I'd love to hear if you have thoughts on --^ and can make changes in another PR if there's a better solution.